### PR TITLE
General update

### PR DESCRIPTION
**Pull request by claudioandre-br**
- Update runtime to 23.08;
- Update pcap to libpcap-1.10.4;
- Update librexgen to 2.1.5;
  - It has been moved to a new maintainer;
- Add a sandbox note to '--list=build-info';
- Start building upstream 39db7dd63e3fefb343c3dbb72eaa5c7599b6c298.
  - We can't keep building the old (and bugged) version forever;

### CODE REVIEW

File: com.openwall.John.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "id": "com.openwall.John",
     "runtime": "org.freedesktop.Platform",
-    "runtime-version": "21.08",
+    "runtime-version": "22.08",
```
Comment (hfiguiere) [inline com.openwall.John.json:4]: 23.08 is out. might as well.

Comment (claudioandre-br) [reply com.openwall.John.json]: Done. I haven't tested it using 23.08 yet; but I will do it using the CI package (if there is any).

**Comment by claudioandre-br** [comment]
The `flatpak install -y --user https://dl.flathub.org/build-repo/50439/com.openwall.John.flatpakref` works fine.


```
Version: 1.9.0-jumbo-1+bleeding-39db7dd63e 2023-09-20 17:02:33 -0300
Build: linux-gnu 64-bit x86_64 AVX2 AC OMP
SIMD: AVX2, interleaving: MD4:3 MD5:3 SHA1:1 SHA256:1 SHA512:1
Deploy: sandboxed as a FlatPak app
System-wide exec: /app/bin
System-wide home: /app/bin
Private home: ~/.john
CPU tests: AVX2
CPU fallback binary: john-xop
OMP fallback binary: john-non-omp
$JOHN is /app/bin/
Format interface version: 14
Max. number of reported tunable costs: 4
Rec file version: REC4
Charset file version: CHR3
CHARSET_MIN: 1 (0x01)
CHARSET_MAX: 255 (0xff)
CHARSET_LENGTH: 24
SALT_HASH_SIZE: 1048576
SINGLE_IDX_MAX: 32768
SINGLE_BUF_MAX: 4294967295
Effective limit: Max. KPC 32768
Max. Markov mode level: 400
Max. Markov mode password length: 30
gcc version: 13.2.0
GNU libc version: 2.38 (loaded: 2.38)
Crypto library: OpenSSL
OpenSSL library version: 030100020
OpenSSL 3.1.2 1 Aug 2023
GMP library version: 6.2.1
Regex library version: 2.1      (loaded: 2.1.5)
File locking: fcntl()
fseek(): fseek
ftell(): ftell
fopen(): fopen
memmem(): System's
times(2) sysconf(_SC_CLK_TCK) is 100
Using times(2) for timers, resolution 10 ms
HR timer: clock_gettime(), latency 62 ns
Total physical host memory: 7795 MiB
Available physical host memory: 7225 MiB
Terminal locale string: C.UTF-8
Parsed terminal locale: UTF-8
```

Testing:
```
--------------------------- test full ---------------------------
Will run 2 OpenMP threads
Testing: descrypt, traditional crypt(3) [DES 256/256 AVX2]... (2xOMP) PASS
[...]
Testing: crypt, generic crypt(3) [?/64]... (2xOMP) PASS
All 423 formats passed self-tests!
 Ok: -test-full=0 --format=cpu

[...]
# More testing
-------------------------------------------
###  Performed 14 tests in 327 seconds  ###
-------------------------------------------
----------------- x86_64 ------------------
```

**Comment by claudioandre-br** [comment]
The "release" will happen in a few days, so, the merge is planned to take place in early October.

**Pull request closed without merging**